src/graphql/schema/Trip/typeDefs.gql

```diff
@@ -4,21 +4,22 @@ extend type Query {
 }
 
 extend type Mutation {
-  createTrip(trip: TripInput!, userID: ID): Trip!
-  addParticipant(tripID: ID, participants: [String]!): Trip!
-  addOrVoteForDestination(tripID: ID, destination: DestinationObjectInput!): Trip!
-  addVoterDestination(tripID: ID, destinationID: ID): Trip!
-  removeVoterDestination(tripID: ID, destinationID: ID): Trip!
-  addTimeFrame(tripID: ID, timeFrame: TimeFrameObjectInput!): Trip!
-  addBudget(tripID: ID, budget: BudgetObjectInput!): Trip!
-  addVoterBudget(tripID: ID, budgetID: ID): Trip!
-  removeVoterBudget(tripID: ID, budgetID: ID): Trip!
-  removeBudget(tripID: ID, budget: BudgetObjectInput!): Trip!
-  voteForDestination(tripID: ID!, destinationID: ID!): Trip!
-  addOrVoteForTimeFrame(tripID: ID!, timeFrames: [TimeFrameInput!]!): [Trip]!
-  addDestination(tripID: ID, destination: DestinationObjectInput!): Trip!
+  createTrip(trip: TripInput!): Trip!
+  addParticipants(tripID: ID, participants: [String!]!): Trip!
+  addOrVoteForDestination(tripID: ID, destinations: [DestinationInput!]!): Trip!
+  addOrVoteForTimeFrame(tripID: ID!, timeFrames: [TimeFrameInput!]!): Trip!
+  addOrVoteForBudget(tripID: ID!, budget: BudgetInput!): Trip!
+  removeVoteForDestination(tripID: ID!, suggestionID: ID!): Trip!
+  removeVoteForTimeFrame(tripID: ID!, suggestionID: ID!): Trip!
+  removeVoteForBudget(tripID: ID!, suggestionID: ID!): Trip!
+  lockDestination(tripID: ID!, suggestionID: ID!): Trip!
+  lockTimeFrame(tripID: ID!, suggestionID: ID!): Trip!
+  lockBudget(tripID: ID!, suggestionID: ID!): Trip!
+  unlockDestination(tripID: ID!): Trip!
+  unlockTimeFrame(tripID: ID!): Trip!
+  unlockBudget(tripID: ID!): Trip!
   leaveTrip(tripID: ID): [Trip]!
-  removeParticipant(tripID: ID, participant: ID): Trip!
+  removeParticipant(tripID: ID, participant: ID): Trip! # TODO allow removal of multiple participants in one query
```

Comment:
Depending on the implementation in the FE. If there is a X on each participant that fires the function this one should be enough. But the way  i started writing the feature was indeed removing an array of participants

Reply:
true, but the way I see it is that if we allow multiple that gives the FE flexibility, if not they're limited.